### PR TITLE
Added timming to seach fields

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
Added trimming to search fields. 
Added checks so that the page will not refresh if the text does not change after refresh

### CODE REVIEW

**Review by Zenen-freshworks — APPROVED** [review]
Looks good and works as intended! We really gotta revisit those filter functions...

**Comment by arranfw** [comment]
This seems to prevent me from adding spaces to filters at all, which could be problematic when searching for names with spaces

image

**Comment by dbayly-freshworks** [comment]
I've removed the trimming onChange. 
Now the only trimming occurs when a query is sent. This should allow for spaces in between words. It will not search if the space added doesn't change the value when trimmed. 
Screen Shot 2021-05-05 at 9 43 52 AM

**Review by arranfw — DISMISSED** [review]
Much better

File: client/src/pages/private/ParticipantTable.js [hunk 1]
```diff
@@ -19,7 +19,8 @@ import {
 import { Table, CheckPermissions, Button, Dialog } from '../../components/generic';
 import { ProspectingForm, InterviewingForm, RejectedForm, HireForm, NewParticipantForm, EditParticipantForm } from '../../components/modal-forms';
 import { useToast } from '../../hooks';
-import { DebounceTextField } from '../../components/generic/DebounceTextField';
+import { DebounceTextField } from '../../components/generic/DebounceTextField'
+;
```
Comment (arranfw) [inline client/src/pages/private/ParticipantTable.js:23]: Extra new-line here

File: client/src/pages/private/ParticipantTable.js [hunk 2]
```diff
@@ -173,27 +173,36 @@ export default () => {
...
       currentPage: 0,
     }));
-    setFsaFilter(value);
+    setFsaFilter(value.trim());
   };
 
   const handleLastNameFilter = (value) => {
-    setPagination(prev => ({
-      ...prev,
+    if(lastNameText && lastNameText.trim() === value.trim()){
```
Comment (arranfw) [inline client/src/pages/private/ParticipantTable.js:187]:
```suggestion
    if(lastNameText?.trim() === value?.trim()){
```

Little bit less wordy option

File: client/src/pages/private/ParticipantTable.js [hunk 3]
```diff
@@ -173,27 +172,33 @@ export default () => {
...
-    setFsaFilter(value);
+    setFsaFilter(value.trim());
   };
 
   const handleLastNameFilter = (value) => {
-    setPagination(prev => ({
-      ...prev,
+    if(lastNameText?.trim() === value?.trim()) return;
+    
+    setPagination(oldPagination => ({
```
Comment (arranfw) [inline client/src/pages/private/ParticipantTable.js:187]: I'm doing a bad job reviewing this... I'm so sorry, these `old____` variables were recently changed to `prev` for consistency, I think this is an unintentional change

Comment (dbayly-freshworks) [reply client/src/pages/private/ParticipantTable.js:187]: All good. I'll make the change.

**Review by arranfw — APPROVED** [review]

**Review by Zenen-freshworks — APPROVED** [review]
nice, this is way more concise

**Pull request merged**